docs/use_cases/batch/part3.md

```diff
@@ -0,0 +1,159 @@
+assign the Delta Lake connector the label *lineitem.historical* and configure
+the datagen conector to **start_after** all historical data has been ingested.
+
+This setup ensures that generated live data starts flowing only when the
+backfill has completed.
+
+
+```json
+CREATE TABLE LINEITEM (...) WITH (
+ 'connectors' = '[{
+    "labels": ["lineitem.historical"],
+    "transport": {
+      "name": "delta_table_input",
+      "config": {
+        "uri": "s3://batchtofeldera/lineitem",
+        "aws_skip_signature": "true",
+        "aws_region": "ap-southeast-2",
+        "mode": "snapshot_and_follow"
+      }
+    }
+ },
+ {
+    "labels": ["lineitem.datagen"],
+    "start_after": "lineitem.historical",
+    "transport": {
+	  "name": "datagen",
+	  "config": {
+            "plan": [{
+              "fields": {
+                "L_PARTKEY": {"strategy": "zipf", "range": [1, 25]},
+                "L_LINENUMBER": {"strategy": "zipf", "range": [1, 25]},
+                "L_RETURNFLAG": {"values": ["A", "N", "R"]},
+                "L_LINESTATUS": {"values": ["F", "0", "F", "F"]},
+                "L_QUANTITY": {"range": [0.01, 99.99]},
+                "L_EXTENDEDPRICE": {"range": [0.02, 99.99]},
+                "L_DISCOUNT": {"range": [0.01, 99.99]},
+                "L_TAX": {"range": [0.01, 99.99]}
+              }
+            }]
+	  }
+    }
+ }]'
+);
+```
+
+## Attempt 1: Using Kafka to Ingest Real Time Data
+
+For a more realistic real-time data ingestion scenario, let's replace the data
+generator with a Kafka based input connector. Here, historical data is first
+loaded from Delta Lake before switching to real time data from Kafka.
+
+By ensuring that Kafka connector **starts after** historical data has been
+ingested, we prevent any duplicate records.
+
+```json
+ {
+    "labels": ["lineitem.live"],
+    "start_after": "lineitem.historical",
+    "transport": {
+        "name": "kafka_input",
+        "config": {
+            "bootstrap.servers": "redpanda:9092",
+            "auto.offset.reset": "latest",
+            "topics": ["lineitem"]
+        }
+    },
+    "format": {
+        "name": "csv",
+        "config": {
+            "delimiter": "|"
+        }
+    }
+ }
+```
+
+### Ensuring Data Consistency with `auto.offset.reset`
+
+In the Kafka connector, we configure **auto.offset.reset** to **latest**. This
+ensures that only the latest messages arriving after the consumer starts are
+ingested, preventing redundancy.
+
+If **auto.offset.reset** is set to **earliest**, all available messages from the
+beginning of the topic, including older historical data, that may have already
+been loaded from Delta Lake, will be processed.
+
+:::danger
+- Ensure all historical data in Kafka is moved to Delta Lake before Feldera
+  starts reading.
+- Failure to do so **may result in data loss** due to missing records not
+  included in the backfill process.
+:::
+
+## Recommended Approach: Kafka and Delta Lake
```

Comment:
So the previous two sections will be gone, and this section will start with a complete table declaration, followed by:

- Explain connector orchestration with labels
- Then explain the more nuanced issue with Kafka offsets.